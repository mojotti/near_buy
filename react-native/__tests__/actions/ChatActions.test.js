import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import {
  createChat,
  handleChatCreation,
  requestChats,
} from '../../src/redux/actions/ChatActions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

const sellerId = 0;
const itemId = 0;
const token = 'fake token';
const title = 'foo';

const ITEM = {
  sellerId,
  itemId,
  title,
};

describe('createChat', () => {
  beforeEach(() => {
    store.clearActions();
  });

  test('creates a new chat', async () => {
    fetch.mockResponseSuccess({ ok: true });

    await await await await store.dispatch(createChat(ITEM, token));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('fails to create a new chat', async () => {
    fetch.mockResponseSuccess({ ok: false });

    await await await await store.dispatch(createChat(ITEM, token));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('throws an error', async () => {
    fetch.mockResponseFailure('error');

    await await await await await store.dispatch(createChat(ITEM, token));
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('requestChats', () => {
  beforeEach(() => {
    store.clearActions();
  });

  test('cannot get chats', async () => {
    fetch.mockResponseSuccess({ chats: 'could not get chats' });

    await await await await store.dispatch(requestChats(token));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('gets chats', async () => {
    fetch.mockResponseSuccess({ chats: [1, 2, 3] });

    await await await await store.dispatch(requestChats(token));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('throws an error', async () => {
    fetch.mockResponseFailure('error');

    await await await await store.dispatch(requestChats(token));
    expect(store.getActions()).toMatchSnapshot();
  });
});
