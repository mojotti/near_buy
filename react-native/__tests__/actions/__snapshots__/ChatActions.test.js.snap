// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createChat creates a new chat 1`] = `
Array [
  Object {
    "type": "CREATE_CHAT_REQUEST",
  },
  Object {
    "type": "CREATE_CHAT_SUCCESSFUL",
  },
  Object {
    "type": "FETCH_CHATS_REQUEST",
  },
]
`;

exports[`createChat fails to create a new chat 1`] = `
Array [
  Object {
    "type": "CREATE_CHAT_REQUEST",
  },
  Object {
    "error": "error",
    "type": "CREATE_CHAT_FAILURE",
  },
]
`;

exports[`createChat throws an error 1`] = `
Array [
  Object {
    "type": "CREATE_CHAT_REQUEST",
  },
  Object {
    "error": "error",
    "type": "CREATE_CHAT_FAILURE",
  },
]
`;

exports[`requestChats cannot get chats 1`] = `
Array [
  Object {
    "type": "FETCH_CHATS_REQUEST",
  },
  Object {
    "error": "could not get chats",
    "type": "FETCH_CHATS_ERROR",
  },
]
`;

exports[`requestChats gets chats 1`] = `
Array [
  Object {
    "type": "FETCH_CHATS_REQUEST",
  },
  Object {
    "chatHeaders": Array [
      1,
      2,
      3,
    ],
    "type": "FETCH_CHATS_SUCCESS",
  },
]
`;

exports[`requestChats throws an error 1`] = `
Array [
  Object {
    "type": "FETCH_CHATS_REQUEST",
  },
]
`;
